_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme

```diff
@@ -86,113 +87,194 @@ function rhdp2_preprocess_node(&$variables) {
 
-    foreach ($authors as $author) {
-      $tids[] = $author['target_id'];
+      if ($read_time[0]['interval'] > 1) {
+        $read_time_text .= 's';
+      }
+
+      $variables['interval'] = $read_time_text ?? '';
     }
+  }
+
+  // Article content type.
+  if ($type == 'article') {
+    $variables['has_hero'] = FALSE;
+    $variables['author_location'] = 'left';
 
-    $author_nodes = Node::loadMultiple($tids);
+    // Set hero and author location if opinion content.
+    $article_type = $node->get('field_article_type')->getValue();
+    if ($article_type && !empty($article_type) && isset($variables['content']['field_image'])) {
+      $article_type_value = $article_type[0]['value'];
 
-    if (count($author_nodes) < 1) {
-      $variables['authors'] = NULL;
+      if ($article_type_value == 'opinion') {
+        $variables['has_hero'] = TRUE;
+        $variables['author_location'] = 'right';
+        $variables['content']['field_image'][0]['#image_style'] = 'hero';
+      }
+    }
+
+    // Process Table of Contents.
+    if ($node->hasField('field_hide_toc')) {
+      $variables['hide_toc'] = $node->field_hide_toc->value ?? FALSE;
     }
-    if (count($author_nodes) === 1) {
-      $nid = reset($author_nodes)->id();
-      $author_url_alias = \Drupal::service('path.alias_manager')->getAliasByPath('/node/'.$nid);
-      $current_path = \Drupal::service('path.current')->getPath();
-      $current_alias = \Drupal::service('path.alias_manager')->getAliasByPath($current_path);
 
-      if ($author_url_alias === $current_alias) {
-        $variables['authors'] = [
-          '#markup' => reset($author_nodes)->getTitle(),
-        ];
+    // Prepares the authors render element.
+    if ($variables['view_mode'] === 'card') {
+      $authors = $node->get('field_author_evangelist')->getValue();
+      $tids = [];
+
+      foreach ($authors as $author) {
+        $tids[] = $author['target_id'];
       }
-      else {
-        // We can reset to retrieve the element since we've already verified there is only 1.
-        $authors_values = [
-          'name' => reset($author_nodes)->getTitle(),
-          // We need a Url object for Drupal, not a plain string.
-          'url' => Url::fromRoute('entity.node.canonical', ['node' => reset($author_nodes)->id()])
-        ];
-
-        // Get a renderable author Link and set the rhd-m-link class.
-        $authors_link = Link::fromTextAndUrl($authors_values['name'], $authors_values['url']);
-        $authors_link = $authors_link->toRenderable();
-        $authors_link['#attributes'] = ['class' => ['rhd-m-link']];
-
-        $variables['authors'] = $authors_link;
+
+      $author_nodes = Node::loadMultiple($tids);
+
+      if (count($author_nodes) < 1) {
+        $variables['authors'] = NULL;
       }
-    }
-    else {
-      $authors_values = [];
 
-      foreach ($author_nodes as $author_node) {
-        $authors_values[] = $author_node->getTitle();
+      if (count($author_nodes) === 1) {
```

Comment:
Could we do `elseif` here?

Reply:
That fixes the @todo - it does not run if there is 0 or 1 authors.

I added multiple authors to a card and it is attached here.  I don't see that's actually in use in production, so I don't think the changes in this PR have altered this output, afaik.

![multiple-authors](https://user-images.githubusercontent.com/19995927/71532602-1d62c480-28c2-11ea-88cc-be661ace3fdd.png)